eos-updater-flatpak-installer/main.c

```diff
@@ -633,6 +783,11 @@ main (int    argc,
             eos_updater_util_filter_for_new_flatpak_ref_actions (flatpak_ref_actions_for_this_boot,
                                                                  flatpak_ref_actions_progress);
 
+          if (!log_all_flatpak_ref_actions ("All flatpak ref actions that are not yet applied",
+                                            new_flatpak_ref_actions_to_apply,
+                                            &error))
```

Comment:
Or here (or anywhere).

Reply:
Yeah, I think this is right. We can change the one point of failure to an assertion and then all this complicated logging failure code can be removed.